datamade/erikson-edi
# Display raw numbers of students surveyed meeting vulnerability thresholds

## Summary

Instead of only displaying percentages of students meeting vulnerability/on track thresholds, display the raw count as well.

Closes #122 

## Demo

Note the new line in the popover displaying `10 students not ready`:

![Screenshot from 2019-03-19 15-57-57](https://user-images.githubusercontent.com/14170650/54638000-60a12480-4a60-11e9-91b4-65b4df3547e2.png)

## Notes

At first, I followed the recommendation in #122 and tried to import these numbers from the spreadsheet. This proved to be prohibitively difficult to do within the time estimate for this feature, since the raw counts are only displayed on Table 3 (the summary table), but we import percentages for each domain using the detail tables that follow.

Instead, I took a shortcut and decided to derive the raw counts based on the percentages. From a quick spot check, this appears to deliver an equivalent feature at much lower cost.

## Testing instructions

- Spin up a development server with `python manage.py runserver`
- Navigate to one (or ideally both) of the community area maps
- Open up the EDI table book in `data/raw/<area>/edi_table_book_2017.xlsx`
- For a handful of domains, choose a neighborhood and spot check that the raw count (`N`) of students meeting vulnerability threshold as reported in Table 3 matches the count displayed on the map
